### PR TITLE
feat(frontend): Allow setting s3 region for artifacts

### PR DESCRIPTION
**Pull request by neerfri**
Allows setting the region of the S3 bucket used by minio to regions other than us-east-1

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by neerfri** [comment]
@googlebot I signed it!

**Comment by neerfri** [comment]
[As it turns out](https://github.com/minio/minio-js/blob/d41c8a806bfe26aa1a618a5a17e812ec8bb54b36/src/main/helpers.js#L89) minio-js uses the `s3.amazonaws.com` to check if the endpoint is really S3, it later chooses the "right" endpoint using the region so I changed the PR to allow setting the region via `AWS_REGION` env var

**Comment by neerfri** [comment]
@Bobgy your review will be appreciated. Thanks!

**Comment by Bobgy** [comment]
/assign @zijianjoy
Can you help with the review?

File: frontend/server/configs.ts [hunk 1]
```diff
@@ -124,6 +125,7 @@ export function loadConfigs(argv: string[], env: ProcessEnv): UIConfigs {
       aws: {
         accessKey: AWS_ACCESS_KEY_ID || '',
         endPoint: 's3.amazonaws.com',
+        region: AWS_REGION || 'us-east-1',
```
Comment (zijianjoy) [inline frontend/server/configs.ts:128]: Does `us-east-1` work if no region is specified? Any documentation we can add here?

**Review by zijianjoy — COMMENTED** [review]
/lgtm

Thank you for the CL! I only have one NIT comment

File: frontend/server/configs.ts (hunk 1 above)
Comment (neerfri) [reply frontend/server/configs.ts:128]: @zijianjoy Yes, minioJS is using the default S3 endpoint (s3.amazonaws.com) which is `us-east-1`

**Comment by neerfri** [comment]
@zijianjoy is there anything still blocking the PR here?
Thanks!

**Comment by zijianjoy** [comment]
/approve

Sorry for the late reply! Looks good to me!

File: frontend/server/configs.ts [hunk 2]
```diff
@@ -58,6 +58,7 @@ export function loadConfigs(argv: string[], env: ProcessEnv): UIConfigs {
     /** minio client use these to retrieve s3 objects/artifacts */
     AWS_ACCESS_KEY_ID,
     AWS_SECRET_ACCESS_KEY,
+    AWS_REGION,
```
Comment (thesuperzapper) [inline frontend/server/configs.ts:61]: @zijianjoy @neerfri I think we also need to retrieve`AWS_DEFAULT_REGION`, which should take precedence over `AWS_REGION` if set.

See AWS docs for `AWS_DEFAULT_REGION` here:
https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html

**Pull request merged**